app/main/profile-command-handler.js

```diff
@@ -72,23 +93,21 @@ export async function handler(
 
     case profileCommandTypes.DID_UNBOOKMARK_LOCATION:
       try {
-        if (!browserWindow || !browserWindow.sessionId) {
-          break;
-        }
-        await storage.starPage(payload.url, browserWindow.sessionId, -1);
+        await storage.starPage(payload.url, payload.sessionId, -1);
         return await dispatchActionsForChangedStarred(payload.url);
       } catch (e) {
         console.log(e); // TODO: do more than ignore failure.
       }
       break;
 
     case profileCommandTypes.DID_SET_USER_TYPED_LOCATION:
-      try {
-        const completions = await storage.visitedMatches(payload.text);
-        return userAgent.set('locations',
-          userAgent.locations.set(payload.text, new Immutable.List(completions)));
-      } catch (e) {
-        console.log(e); // TODO: do more than ignore failure.
+      if (respond) {
```

Comment:
This would be a weird scenario if the frontend sent this action without a respond callback.  I don't know that it's worth throwing, but maybe at least warning (unless if you can think of a reason this should happend)